clients/src/main/java/org/apache/kafka/common/protocol/Writable.java

```diff
@@ -33,6 +35,23 @@
     void writeVarint(int i);
     void writeVarlong(long i);
 
+    default void writeApiMessage(
+        ApiMessage message,
+        ObjectSerializationCache serializationCache,
+        short version
+    ) {
+        message.write(this, serializationCache, version);
+    }
+
+    default void writeRecords(BaseRecords records) {
```

Comment:
Does it need null check (maybe no-op)?

Reply:
I was concerned about this also, but the generated code adds its own null check.